Content/Items/Lightsaber/LightsaberExtras.cs

```diff
@@ -0,0 +1,283 @@
+﻿using StarlightRiver.Helpers;
+using System;
+using System.Collections.Generic;
+using Terraria.DataStructures;
+using Terraria.Graphics.Effects;
+using Terraria.ID;
+
+namespace StarlightRiver.Content.Items.Lightsaber
+{
+	public class LightsaberGProj : GlobalProjectile
```

Comment:
This seems like perhaps we could have a more general solution for this? Need to evaluate

Reply:
I'll leave it alone for now because I'm not sure about that. I guess you or someone else in the dev team can help with the general solution?